### DISCUSSION TITLE
[Feature Request] Applying TailwindCSS breakpoints to the `size` prop

### DISCUSSION THREAD

**Discussion by gusapps** [heroui-inc/heroui#4732] [Feature requests]
### Is your feature request related to a problem? Please describe.

Sometimes you want a component, such as a`Button`, to be size `md` on desktop and `sm` on mobile.

### Describe the solution you'd like

One API proposal is for the `size` prop to accept an object with Tailwind breakpoints as keys:
```tsx
<Button
  size={{
    sm: "sm",
    md: "sm",
  }}
/>
```

### Describe alternatives you've considered

An alternative API design is a `breakpointsProps` prop:

```tsx
<Button
  size="md"
  breakpointsProps={{
    sm: { size: "sm" },
  }}
/>
```

**Comment by linear** [comment]
<p><a href="https://linear.app/nextui-inc/issue/ENG-1394/[feature-request]-applying-tailwindcss-breakpoints-to-the-size-prop">ENG-1394 [Feature Request] Applying TailwindCSS breakpoints to the `size` prop</a></p>
